apis/network/v1beta1/ipset_types.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+)
+
+// IPSetNetwork Type
+type IPSetNetwork struct {
+	// +kubebuilder:validation:Required
+	// Network Name
+	Name NetNameStr `json:"name"`
+
+	// +kubebuilder:validation:Required
+	// Subnet Name
+	SubnetName NetNameStr `json:"subnetName"`
+
+	// +kubebuilder:validation:Optional
+	// Fixed Ip
+	// TODO: add validation to webhook
+	FixedIP *string `json:"fixedIP,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// Use gateway from subnet as default route. There can only be one default route defined per IPSet.
+	DefaultRoute *bool `json:"defaultRoute,omitempty"`
+}
+
+// IPSetSpec defines the desired state of IPSet
+type IPSetSpec struct {
+	// Hostname
+	Hostname *HostNameStr `json:"hostname"`
```

Comment:
Do we need this? Won't the requester be the owner of this?

Reply:
right now we use the hostname to validate for an already reserved IP. you suggest we drop the hostname from ipam at all and use the IPSetRef for this?

Reply:
Yeah I meant that, reservation is for a IPSet, whether it's created by dataplane role/node or a controlplane service.

Reply:
yes, I think that makes sense. the IPSetRef is a clear reference to the IPSet and not having the hostname is more generic and the requestor handles it.

Reply:
done, removed hostname